### ISSUE TITLE
CachyOSv3 and v4 Mirrorlists

### ISSUE THREAD

**Issue by KenJyn76** [westandskif/rate-mirrors#40]
Hi! I was hoping it was possible to add flags for the CachyOS v3 and v4 mirrorlists. They use the same servers, with the only difference being the location of the repo, in .../repo/$archv3 or $archv4 instead of $arch. Currently I'm just going to

`sed /s/$arch/$archv3 cachyos-v3-mirrorlist` 

but it would be nice if this is possible to conveniently implement.

Thanks

**Comment by westandskif** [comment]
Hi! Sure, I'll get back to it one of these days.

**Comment by ptr1337** [comment]
Great idea at all.
Currently we use this script to rank our the mirrors, including arch ones.
https://github.com/CachyOS/CachyOS-PKGBUILDS/blob/master/cachyos-rate-mirrors/cachyos-rate-mirrors

But a native implementation would be good. :)

**Comment by vnepogodin** [comment]
all CachyOS repos are supported
like that we use in CachyOS:
V3
```
rate-mirrors cachyos --arch x86_64_v3
```
V4
```
rate-mirrors cachyos --arch x86_64_v4
```

**Comment by vnepogodin** [comment]
It's actually completely useless to have `cachyos-v3` `cachyos-v4` entries, if the only difference is arch

**Comment by westandskif** [comment]
Hi guys! I was looking at the script - [cachyos-rate-mirrors](https://github.com/CachyOS/CachyOS-PKGBUILDS/blob/master/cachyos-rate-mirrors/cachyos-rate-mirrors) when you replied with it :) It looks like you are all set with it.

Probably the only missing thing is auto-detection of the arch, but I'm not sure whether it's of any interest for @ptr1337 and @vnepogodin to move this logic to `rate-mirrors`. I believe it makes sense to keep it on `cachyos-rate-mirrors` side for the sake of flexibility (_e.g. any changes to supported CPU instructions would be kept within the bash script managed by CachyOS, while the simple part of rating mirrors would be taken care of by `rate-mirrors`_).

Either way, as long as `rate-mirrors` binary is kept small and one distro doesn't dramatically affect others, I defer to @ptr1337 and @vnepogodin on this.

**Comment by ptr1337** [comment]
Yes, maybe a additional architecture check would be a idea at all. Right now this is doing our:
- Installers
-  The "repo-add" script for adding the repos

We will look into that. Generally I think it is more smart, to simply rate all arches and installed all of these mirrorslists.
Adding the repos to pacman is handeled by the scripts as written above.

@vnepogodin

**Comment by KenJyn76** [comment]
Thanks for the feedback, guys. As of now, I'm using this alias in my .zshrc in order to get everything updated at once.

`reflector() {

[ -d /etc/pacman.d/backup ] || sudo mkdir /etc/pacman.d/backup; \
[ ! -r /etc/pacman.d/mirrorlist ] || sudo mv /etc/pacman.d/mirrorlist /etc/pacman.d/backup/mirrorlist; \
[ ! -r /etc/pacman.d/endeavouros-mirrorlist ] || sudo mv /etc/pacman.d/endeavouros-mirrorlist /etc/pacman.d/backup/endeavouros-mirrorlist; \
[ ! -r /etc/pacman.d/cachyos-mirrorlist ] || sudo mv /etc/pacman.d/cachyos-mirrorlist /etc/pacman.d/backup/cachyos-mirrorlist; \
[ ! -r /etc/pacman.d/cachyos-v3-mirrorlist ] || sudo mv /etc/pacman.d/cachyos-v3-mirrorlist /etc/pacman.d/backup/cachyos-v3-mirrorlist; \

export TMPFILE="$(mktemp)"; \
    sudo true; \
    rate-mirrors --save=$TMPFILE arch --max-delay=43200 \
      && sudo mv $TMPFILE /etc/pacman.d/mirrorlist

export TMPFILE="$(mktemp)"; \
    sudo true; \
    rate-mirrors --save=$TMPFILE endeavouros \
      && sudo mv $TMPFILE /etc/pacman.d/endeavouros-mirrorlist

export TMPFILE="$(mktemp)"; \
    sudo true; \
    rate-mirrors --save=$TMPFILE cachyos \
      && sudo mv $TMPFILE /etc/pacman.d/cachyos-mirrorlist \
      && sed 's/$arch/$arch_v3/' /etc/pacman.d/cachyos-mirrorlist | sudo tee /etc/pacman.d/cachyos-v3-mirrorlist

}`

I'm not sure it's useful for development purposes, but someone may be able to make use of it.

My CPU doesn't have x86_64_v4 support, but I noticed that CachyOS installed its v4 mirrorlist to my PC anyway, and simply did not enable it

**Comment by westandskif** [comment]
Hi guys, thanks for sharing! It really feels like `cachyos-rate-mirrors` script is the right place to identify architecture without delegating it to 3rd party tools. So, I'm closing this issue until we have more input on this.
